separate release job in the workflow

To avoid situation like https://github.com/bahmutov/npm-install/runs/453466017?check_suite_focus=true

```
Run cycjimmy/semantic-release-action@v2
Error: Command failed: npm ci --only=prod
npm ERR! code EACCES
npm ERR! syscall mkdir
npm ERR! path /home/path
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, mkdir '/home/path'
npm ERR!  [OperationalError: EACCES: permission denied, mkdir '/home/path'] {
npm ERR!   cause: [Error: EACCES: permission denied, mkdir '/home/path'] {
npm ERR!     errno: -13,
npm ERR!     code: 'EACCES',
npm ERR!     syscall: 'mkdir',
npm ERR!     path: '/home/path'
npm ERR!   },
npm ERR!   isOperational: true,
npm ERR!   stack: "Error: EACCES: permission denied, mkdir '/home/path'",
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'mkdir',
npm ERR!   path: '/home/path'
npm ERR! }
```

the release went wrong because one of the tests allowed `core.exportVariable('npm_config...')` to happen, changing the environment. If we separate the release job into its own container, the tests will never affect it.

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -27,7 +27,7 @@ jobs:
     name: release
     needs: build-and-test
     # only release from the master branch
-    if: github.ref == 'master'
+    # if: github.ref == 'master'
     steps:
       - name: Checkout
         uses: actions/checkout@v1

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -32,7 +32,7 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v1
         # can we use THIS action right away?!
-      - uses: bahmutov/npm-install@ ${{ github.sha }}
+      - uses: bahmutov/npm-install@$GITHUB_SHA
 
       # https://github.com/cycjimmy/semantic-release-action
       - name: Semantic Release

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -2,32 +2,32 @@ name: main
 on: [push]
 
 jobs:
-  # build-and-test:
-  #   runs-on: ubuntu-latest
-  #   name: Build and test
-  #   steps:
-  #     - name: Checkout
-  #       uses: actions/checkout@v1
+  build-and-test:
+    runs-on: ubuntu-latest
+    name: Build and test
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
 
-  #     # caching NPM dependencies
-  #     # https://github.com/actions/cache/blob/master/examples.md#node---npm
-  #     - uses: actions/cache@v1
-  #       with:
-  #         path: ~/.npm
-  #         key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
-  #         restore-keys: |
-  #           ${{ runner.os }}-node-
+      # caching NPM dependencies
+      # https://github.com/actions/cache/blob/master/examples.md#node---npm
+      - uses: actions/cache@v1
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
 
-  #     - run: node -v
-  #     - run: npm ci
-  #     - run: npm test
+      - run: node -v
+      - run: npm ci
+      - run: npm test
 
   release:
     runs-on: ubuntu-latest
     name: release
-    # needs: build-and-test
+    needs: build-and-test
     # only release from the master branch
-    # if: github.ref == 'master'
+    if: github.ref == 'master'
     steps:
       - name: Checkout
         uses: actions/checkout@v1

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -32,7 +32,7 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v1
         # can we use THIS action right away?!
-      - uses: bahmutov/npm-install@${{ github.sha }}
+      - uses: bahmutov/npm-install@ ${{ github.sha }}
 
       # https://github.com/cycjimmy/semantic-release-action
       - name: Semantic Release

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -31,8 +31,13 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v1
-        # can we use THIS action right away?!
+
+      # trying to use this action to install dependencies for semantic release
       - uses: bahmutov/npm-install@HEAD
+
+      - run:
+          name: print GitHub SHA
+          command: npx @bahmutov/print-env GITHUB_SHA
 
       # https://github.com/cycjimmy/semantic-release-action
       - name: Semantic Release

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -22,6 +22,18 @@ jobs:
       - run: npm ci
       - run: npm test
 
+  release:
+    runs-on: ubuntu-latest
+    name: release
+    needs: build-and-test
+    # only release from the master branch
+    if: ${{ github.ref }} == master
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+        # can we use THIS action right away?!
+      - uses: bahmutov/npm-install@${{ github.sha }}
+
       # https://github.com/cycjimmy/semantic-release-action
       - name: Semantic Release
         uses: cycjimmy/semantic-release-action@v2

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -32,7 +32,7 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v1
         # can we use THIS action right away?!
-      - uses: bahmutov/npm-install@${{github.sha}}
+      - uses: bahmutov/npm-install@HEAD
 
       # https://github.com/cycjimmy/semantic-release-action
       - name: Semantic Release

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -27,7 +27,7 @@ jobs:
     name: release
     needs: build-and-test
     # only release from the master branch
-    if: ${{ github.ref }} == master
+    if: github.ref == master
     steps:
       - name: Checkout
         uses: actions/checkout@v1

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -32,11 +32,10 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v1
 
+      - run: echo $GITHUB_SHA
+
       # trying to use this action to install dependencies for semantic release
       - uses: bahmutov/npm-install@HEAD
-
-      - name: print GitHub SHA
-        run: npx @bahmutov/print-env GITHUB_SHA
 
       # https://github.com/cycjimmy/semantic-release-action
       - name: Semantic Release

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -35,9 +35,8 @@ jobs:
       # trying to use this action to install dependencies for semantic release
       - uses: bahmutov/npm-install@HEAD
 
-      - run:
-          name: print GitHub SHA
-          command: npx @bahmutov/print-env GITHUB_SHA
+      - name: print GitHub SHA
+        run: npx @bahmutov/print-env GITHUB_SHA
 
       # https://github.com/cycjimmy/semantic-release-action
       - name: Semantic Release

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -27,7 +27,7 @@ jobs:
     name: release
     needs: build-and-test
     # only release from the master branch
-    if: github.ref == master
+    if: github.ref == 'master'
     steps:
       - name: Checkout
         uses: actions/checkout@v1

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -2,37 +2,37 @@ name: main
 on: [push]
 
 jobs:
-  build-and-test:
-    runs-on: ubuntu-latest
-    name: Build and test
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v1
+  # build-and-test:
+  #   runs-on: ubuntu-latest
+  #   name: Build and test
+  #   steps:
+  #     - name: Checkout
+  #       uses: actions/checkout@v1
 
-      # caching NPM dependencies
-      # https://github.com/actions/cache/blob/master/examples.md#node---npm
-      - uses: actions/cache@v1
-        with:
-          path: ~/.npm
-          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
-          restore-keys: |
-            ${{ runner.os }}-node-
+  #     # caching NPM dependencies
+  #     # https://github.com/actions/cache/blob/master/examples.md#node---npm
+  #     - uses: actions/cache@v1
+  #       with:
+  #         path: ~/.npm
+  #         key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+  #         restore-keys: |
+  #           ${{ runner.os }}-node-
 
-      - run: node -v
-      - run: npm ci
-      - run: npm test
+  #     - run: node -v
+  #     - run: npm ci
+  #     - run: npm test
 
   release:
     runs-on: ubuntu-latest
     name: release
-    needs: build-and-test
+    # needs: build-and-test
     # only release from the master branch
     # if: github.ref == 'master'
     steps:
       - name: Checkout
         uses: actions/checkout@v1
         # can we use THIS action right away?!
-      - uses: bahmutov/npm-install@$GITHUB_SHA
+      - uses: bahmutov/npm-install@github.sha
 
       # https://github.com/cycjimmy/semantic-release-action
       - name: Semantic Release

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -32,7 +32,7 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v1
         # can we use THIS action right away?!
-      - uses: bahmutov/npm-install@github.sha
+      - uses: bahmutov/npm-install@${{github.sha}}
 
       # https://github.com/cycjimmy/semantic-release-action
       - name: Semantic Release

--- a/package.json
+++ b/package.json
@@ -31,6 +31,7 @@
     "@actions/core": "1.2.0",
     "@actions/exec": "1.0.1",
     "@actions/io": "1.0.1",
+    "@bahmutov/print-env": "1.2.0",
     "cache": "github:cypress-io/github-actions-cache#8bec6cc",
     "hasha": "5.1.0",
     "quote": "0.4.0"

